### PR TITLE
Sketchify your Photo

### LINKED ISSUE #80: Sketchify your Photo Microbyte
**MicroByte Title:** Sketchify your Photo Microbyte

**Domain:** Python, ML, Open CV

**Scope of learning for learners:** Understand and implement basics of open cv with a real-time algorithm that the photo editor app uses for making a pencil sketch.

**Activities:** The activity will include a pencil sketch generation from a photo using fundamentals of open cv.

### PR DESCRIPTION
**Pull request by rutujak24**
## Closes #80 

<Put a x between the square brackets. Make sure you get them all>

### Checklist: 

- [x] Added activities for learners
- [x] Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x] Clean and well documented code (if source code files are involved)
- [x] Added images for better understanding if deemed necessary

### CODE REVIEW

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @rutujak24!
Just reviewed your PR. It is good content but it is not put around well.
I would request you to make a few changes -

1. We expect the readme to have significantly more detail to it than just want you have provided. By this I mean that you need to specify the goals more clearly (When you say sketchify, what kind of sketch? - Black and White, Sepia, Color? I hope you get my point)
At least a few more lines of detail are required.

2. The structure of the initial part is a bit off:
image
Include the intro and first in a ROWwise fashion according to point 1 and then add your images again in a row-wise manner, keeping it like this is completely displeasing to read as a learner. Then, bring on the prerequisites and the activities.

3. I personally think it is not a good idea to use personal images for project purposes due to some privacy concerns. I suggest you to do the same. Use a non-personal image for project purposes, not just now but for your future projects too. (Take it down from both images folder as well as README)

4. The activity description you have given in README is zero. You need to include some info regarding what will they get if they go to src/ and then that activity. Kindly include a short intro about each activity in its respective place.

5. Activity 1, 2, and 3 are mostly just one cell-based code, check if it is possible to add something more. (Some reference links that are not straightly pointing to the solution as you have given in each of these notebooks will do)
6. Do not provide the solution to Activity 5 since it can be used as a learner activity later. (You can rename it to 'Test Yourself' if you like too)

Good topic selection this time.
Kindly make these changes asap and we shall re review it asap.
All the best!

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hi @rutujak24!
I reviewed it and this is a great development over the older version. I really like how it looks right now.
I just have a few small changes to suggest to improve it a bit more and we are good to go after that:

1. In Activity 2 and 3, you have just given a single cell to fill all the code in. Kindly break it down into multiple cells so that a learner doesn't get confused by thinking that one cell = less code or something such.

2. You can rename `Activity 5` to `Test Yourself` or something on similar lines.

3. This is not a change but more like an observation by me - why is there so much extra space in the links provided in the notebook files? (Highlighted in screenshot attached below) I am not sure if it is by design or not. If it is possible to correct it, kindly do it. (True for all notebooks from Activity 1 - 5)
image

Thanks for your prompt response.
Make these changes and we shall re-review it asap.

**Review by amoghrajesh — APPROVED** [review]
Hey @rutujak24!
Thanks for your quick response. I really appreciate it.
Great work on the changes and I have no further complaints about it now.

Till the other reviewers get on it, try to think of a few learner issues that can be raised if possible scope wise or any improvement a learner can work on.
Thanks for your contribution. 
Approving it!

File: Sketchify your Photo/README.md
Comment (YourAKShaw) [inline Sketchify your Photo/README.md]: `generate an automatic pencil sketch`

I dunno what's with me and grammar. :sweat_smile:

**Review by YourAKShaw — APPROVED** [review]
@rutujak24 I've got no complaints this time. :+1: 

Maybe because @amoghrajesh kinda addressed the issues before I could get to them, and you've done a fine resolution too.

Just get over with this small request of a commit and we'll merge this PR.

**Review by Kevinpaulose05 — CHANGES_REQUESTED** [review]
Nice work overall. 
One big improvement I just observed was that u should give out the final output of each activity, how the image or value of the variable should be. If possible mention these outputs in the README also. Visual images are eye-catching elements. Like how the cat will look after a negative of the image and gaussian blur. 

Also kindly check these:
1. In pre-requisites mention the environment preferred to be working on is Jupyter. Something like pre-requisites of [this](https://github.com/rutujak24/Miscellaneous/tree/main/COVID-19%20Visualization%20using%20Pandas%20and%20Matplotlib). 
2. There is some formatting issue in "Further Read" section of activities 1 & 2. Correct those.  @amoghrajesh has already pointed this out

**Comment by Kevinpaulose05** [comment]
> Nice work overall.
> One big improvement I just observed was that u should give out the final output of each activity, how the image or value of the variable should be. If possible mention these outputs in the README also. Visual images are eye-catching elements. Like how the cat will look after a negative of the image and gaussian blur.
> 
> Also kindly check these:
> 
> 1. In pre-requisites mention the environment preferred to be working on is Jupyter. Something like pre-requisites of [this](https://github.com/rutujak24/Miscellaneous/tree/main/COVID-19%20Visualization%20using%20Pandas%20and%20Matplotlib).
> 2. There is some formatting issue in "Further Read" section of activities 1 & 2. Correct those.  @amoghrajesh has already pointed this out

Kindly let us know when these r done @rutujak24

**Comment by rutujak24** [comment]
> > Nice work overall.
> > One big improvement I just observed was that u should give out the final output of each activity, how the image or value of the variable should be. If possible mention these outputs in the README also. Visual images are eye-catching elements. Like how the cat will look after a negative of the image and gaussian blur.
> > Also kindly check these:
> > 
> > 1. In pre-requisites mention the environment preferred to be working on is Jupyter. Something like pre-requisites of [this](https://github.com/rutujak24/Miscellaneous/tree/main/COVID-19%20Visualization%20using%20Pandas%20and%20Matplotlib).
> > 2. There is some formatting issue in "Further Read" section of activities 1 & 2. Correct those.  @amoghrajesh has already pointed this out
> 
> Kindly let us know when these r done @rutujak24

I have just pushed the changes, please have a look.

**Review by Kevinpaulose05 — APPROVED** [review]
Well done. I liked this and even u can now understand what he previous MB of urs was lacking in. Anyways I just have one last small request. Please add the photos in the README too after each activity. It will make ur MB more attractive.

Two things to be done:
1. Open issues for learners
2. Showcase

Cheers

**Comment by YourAKShaw** [comment]
@rutujak24 The third image seems wrong. Kindly fix that.

**Comment by rutujak24** [comment]
> @rutujak24 The third image seems wrong. Kindly fix that.

done

**Comment by YourAKShaw** [comment]
@rutujak24 Kindly showcase this MicroByte in the **#microbyte-showcase** channel over at our slack workspace, using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) template for the same.

**Pull request merged**
